src/connector.rs

```diff
@@ -305,9 +297,12 @@ impl EthConnectorContract {
         }
     }
 
-    /// Internal ETH deposit logic
-    pub(crate) fn internal_deposit_eth(&mut self, address: &Address, amount: &U256) {
-        self.ft.internal_deposit_eth(address.0, amount.as_u128());
+    /// Internal logic for explicitly setting an eth balance (needed by ApplyBackend for Engine)
+    pub(crate) fn internal_set_eth_balance(&mut self, address: &Address, amount: &U256) {
+        // TODO: this `as_u128` looks pretty dangerous! We need to somehow handle the case where
```

Comment:
In theory, it should be impossible I think. I think that would be much more ETH than there is in existence and may end up being a pointless test. In virtually every other case, I would agree, it is dangerous, but I would argue it really isn't in this instance since it should be impossible to happen and if it does, it would be a much larger issue with the engine itself.

Reply:
The problem is that we use the same type for `balances` and `total_supply`. But we discussed it with @djsatok and decided that situation (more > u128) probably impossible. But we should keep it in mind.